appinventor/components/src/com/google/appinventor/components/runtime/ChartDataBase.java

```diff
@@ -127,6 +135,84 @@ public void Clear() {
         refreshChart();
     }
 
+
+    /**
+     * Imports data from a CSV file component, with the specified column names.
+     *
+     * TODO: Support multi-dimension data components
+     *
+     * @param csvFile  CSV File component to import form
+     * @param xValueColumn  x-value column name
+     * @param yValueColumn  y-value column name
+     */
+    @SimpleFunction(description = "Imports data from the specified CSVFile component, given the names of the " +
+            "X and Y value columns. Passing in empty text for any of the column parameters will result" +
+            " in the usage of the default option of entry 1 having the value of 0, entry 2 having the value of" +
+            " 1, and so forth.")
+    public void ImportFromCSV(final CSVFile csvFile, String xValueColumn, String yValueColumn) {
+        // Construct a YailList of columns from the specified parameters
+        final YailList columns = YailList.makeList(Arrays.asList(xValueColumn, yValueColumn));
+
+        // Import the data from the CSV file with the specified columns asynchronously
+        AsynchUtil.runAsynchronously(new Runnable() {
+            @Override
+            public void run() {
+                // Import from CSV file with the specified parameters
+                chartDataModel.importFromCSV(csvFile, columns);
+
+                // Update the UI after importing (must be done on UI
+                // thread to avoid exceptions)
+                container.$context().runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        refreshChart();
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * Sets the CSV columns to parse data from the CSV source.
+     *
+     * TODO: Hide property in case the Source is not a CSVFile.
+     *
+     * @param columns  CSV representation of the column names (e.g. A,B will
+     *                 use A for the x values, and B for the y values)
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING, defaultValue = "")
```

Comment:
Look into writing a property setter here.

Either implement it in this PR, or create an issue.

Reply:
TBD in #95